From 5ec517ec7ba6ba044f1b98681613bdffb42dca33 Mon Sep 17 00:00:00 2001
From: Poovendhan Selvaraj <quic_poovendh@quicinc.com>
Date: Tue, 3 Sep 2024 18:32:44 +0530
Subject: [PATCH] firmware: scm: Add support to check if fuse row write scm are
 available

Add an interface to check if fuse row write scm is supported or not.

Change-Id: I1b4c52b8ce28c05b92e55c3e353bd45d7d03d329
Signed-off-by: Poovendhan Selvaraj <quic_poovendh@quicinc.com>
---
 drivers/firmware/qcom_scm.c            | 13 +++++++++++++
 include/linux/firmware/qcom/qcom_scm.h |  1 +
 2 files changed, 14 insertions(+)

diff --git a/drivers/firmware/qcom_scm.c b/drivers/firmware/qcom_scm.c
index 3e9209fb1964..e3a64d289638 100644
--- a/drivers/firmware/qcom_scm.c
+++ b/drivers/firmware/qcom_scm.c
@@ -1465,6 +1465,19 @@ bool qcom_sec_dat_fuse_available(void)
 }
 EXPORT_SYMBOL_GPL(qcom_sec_dat_fuse_available);
 
+/**
+ * qcom_qfrom_fuse_row_write_available() - is the fuse row write interface
+ *                                         available ?
+ *
+ * Return: true if the SCM call is supported
+ */
+bool qcom_qfrom_fuse_row_write_available(void)
+{
+	return __qcom_scm_is_call_available(__scm->dev, QCOM_SCM_SVC_FUSE,
+					    QCOM_QFPROM_ROW_WRITE_CMD);
+}
+EXPORT_SYMBOL_GPL(qcom_qfrom_fuse_row_write_available);
+
 /**
  * qcom_scm_ice_invalidate_key() - Invalidate an inline encryption key
  * @index: the keyslot to invalidate
diff --git a/include/linux/firmware/qcom/qcom_scm.h b/include/linux/firmware/qcom/qcom_scm.h
index c727e0bd572d..e8e3c6b78847 100644
--- a/include/linux/firmware/qcom/qcom_scm.h
+++ b/include/linux/firmware/qcom/qcom_scm.h
@@ -294,6 +294,7 @@ extern bool qcom_scm_ice_available(void);
 extern bool qcom_scm_ice_hwkey_available(void);
 extern bool qcom_qfprom_show_auth_available(void);
 extern bool qcom_sec_dat_fuse_available(void);
+extern bool qcom_qfrom_fuse_row_write_available(void);
 extern int qcom_scm_ice_invalidate_key(u32 index);
 extern int qcom_config_sec_ice(void *buf, int size);
 extern int qcom_scm_ice_set_key(u32 index, const u8 *key, u32 key_size,
-- 
2.34.1

